android/src/toga_android/widgets/box.py

```diff
@@ -24,6 +23,8 @@ def set_child_bounds(self, widget, x, y, width, height):
         self.native.updateViewLayout(widget.native, layout_params)
 
     def set_background_color(self, value):
-        self.native.setBackgroundColor(
-            native_color(TRANSPARENT if (value is None) else value)
-        )
+        self.set_background_color_simple(value)
+
+    def rehint(self):
+        self.interface.intrinsic.width = at_least(0)
+        self.interface.intrinsic.height = at_least(0)
```

Comment:
I've added this implementation to Android, iOS and Winforms, which previously had no `rehint` method on `Box`, either directly or inherited from `Widget`. This was causing a box without an explicit size to become as large as possible, completely messing up even fairly simple layouts like the one in example/resize. (Previously noted [here](https://github.com/beeware/toga/issues/1502#issuecomment-1154259939).)

Reply:
That seems odd - rehint() is defined on dummy/base.py, so every base widget *must* provide a rehint method - the iOS, Winforms and Android ones were just no-ops. By *not* defining an intrinsic size, the widget is definitely going to use all available space - but then, that *should* be same behavior that `at_least(0)` should provide, so I'm a little unclear *why* this changes anything.

Reply:
As noted in a follow up comment - there's a subtle difference in the Pack implementation in how space is allocated to children, depending on whether the child has an intrinsic size of `at_least(0)` and `None`. `at_least(0)` is a correct definition, and seems to generate the intended behaviour. There's possible a bug that this *should* be how Pack operates when there's no intrinsic size defined, but this at least gives us the right behavior for widgets right now.